typing/typecore.ml

```diff
@@ -5089,6 +5139,49 @@ and type_andops env sarg sands expected_ty =
   let let_arg, rev_ands = loop env sarg (List.rev sands) expected_ty in
   let_arg, List.rev rev_ands
 
+(*TODO mbungeroth: Why do I need the type annotation here to make this work?*)
+  and comprehension ~loc ~env ~ty_expected ~container_type 
+                  ~sbody ~(comp_typell : Parsetree.comprehension list list) = 
+
+  let from_to_comprehension ~body_env ~env param slow shigh dir =
```

Comment:
`from_to_comprehension` and `in_comprehension` can probably just be inlined into `call`. I'd also probably lift `call` out of `comprehension` and call it something like `type_comprehension_clause`.

Reply:
Good question. I think that once a function gets to a certain size it is better to lift it out. For example, `type_function` is only called from `type_expect`, but having it as a local function in there would make things much harder to read. I also think it is slightly better to lift things out if they make semantic sense on their own.

I think both those points are debatable here, so it's reasonable to do it either way.